src/hw_14.ts

```diff
@@ -0,0 +1,200 @@
+//Кожний нотаток має назву, зміст, дату створення і редагування та статус. Нотатки бувають двох типів.
+//Дефолтні та такі, які вимагають підтвердження при ридагуванні.
+enum TaskStatusEnum {
+  DEFAULT,
+  CONFIRM,
+}
+type TaskType = {
+  name: string;
+  description: string;
+  status: TaskStatusEnum;
+};
+
+class Task {
+  public readonly id: number;
+  public readonly createdDate: Date;
+  public updatedDate: Date;
+  public name: string;
+  public description: string;
+  public status: TaskStatusEnum;
+  public completed: boolean;
+  constructor(task: TaskType) {
+    if (!task.name) throw new Error(`Field "Name" cannot be empty`);
+    if (!task.description)
+      throw new Error(`Field "description" cannot be empty`);
+    this.id = Math.random();
+    this.name = task.name;
+    this.description = task.description;
+    this.status = task.status;
+    this.createdDate = new Date();
+    this.updatedDate = new Date();
+    this.completed = false;
+  }
+  public update(description: string): Task {
+    this.confirmed();
+    this.description = description;
+    this.updatedDate = new Date();
+    return this;
+  }
+  public markCompleted(): Task {
+    this.confirmed();
+    this.completed = true;
+    this.updatedDate = new Date();
+    return this;
+  }
+  private confirmed(): void {
+    if (this.status === TaskStatusEnum.CONFIRM) {
+      console.log(`Confirm editing note!`);
+    }
+  }
+}
+//Вам необхідно написати додаток Todo list.
+//У списку нотаток повинні бути методи для додавання нового запису, видалення, редагування та отримання повної інформації про нотатку за ідентифікатором,
+//а так само отримання списку всіх нотаток.
+abstract class App {
+  /**
+   * addTask
+   */
+  public add(name: string, description: string, status: TaskStatusEnum) {}
+  /**
+   * remove
+   */
+  public remove(id: number) {}
+  /**
+   * edit
+   */
+  public edit(id: number, description: string) {}
+  /**
+   * getAllTasks
+   */
+  public getAllTasks() {}
+  /**
+   * getTaskById
+   */
+  public getTaskById(id: number) {}
+
+  /**
+   * getCountTask
+   */
+  public getCountTask() {}
+  /**
+   * uncomplitedCount
+   */
+  public uncomplitedCount() {}
+  /**
+   * markCompleted
+   */
+  public markCompleted(id: number) {}
+}
+
+interface IAdditionalMethods {
+  searchTask(query: string): Task[];
```

Comment:
The task requirements sound like:
"Separately, it is necessary to expand the behavior of the list and add the possibility of searching for a note by name or content."

It means, that the `searchTask ` method should be a part of a new class, that will expand the `TodoList` class.